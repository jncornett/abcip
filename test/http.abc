c ( b=80 )

a ( syn )
b ( syn, ack )
a ( ack )

a ( ack, data="GET / HTTP/1.1|d a d a|" )

b ( ack, data=
    "HTTP/1.1 200 OK|d a|"
    "Transfer-Encoding: chunked|d a|"
    "Content-Type: text/html; charset=iso-8859-1|d a d a|"
)

b ( ack, data="A|d a|ABCDEFGHIJ|d a|" )
b ( ack, data="4|d a|EFGH|d a|" )
a ( ack )

b ( ack, data="4|d a|IJKL|d a|" )
b ( ack, data="4|d a|MNOP|d a d a|" )
a ( ack )

b ( fin, ack )
a ( fin, ack )
b ( ack )

