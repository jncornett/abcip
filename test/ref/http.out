IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [S], cksum 0xb391 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0001 0000 0000 5002  .....P........P.
	0x0030:  2000 b391 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [S.], cksum 0xb37f (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0050 bdec 0000 0001 0000 0002 5012  ...P..........P.
	0x0030:  2000 b37f 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xb380 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 b380 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 58)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xd4cd (correct), seq 1:19, ack 1, win 8192, length 18
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  003a 0003 0000 4006 5ca8 0a01 0203 0a09  .:....@.\.......
	0x0020:  0807 bdec 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 d4cd 0000 4745 5420 2f20 4854 5450  ......GET./.HTTP
	0x0040:  2f31 2e31 0d0a 0d0a                      /1.1....
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 132)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0x2dfc (correct), seq 1:93, ack 19, win 8192, length 92
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0084 0002 0000 4006 5c5f 0a09 0807 0a01  ......@.\_......
	0x0020:  0203 0050 bdec 0000 0002 0000 0014 5010  ...P..........P.
	0x0030:  2000 2dfc 0000 4854 5450 2f31 2e31 2032  ..-...HTTP/1.1.2
	0x0040:  3030 204f 4b0d 0a54 7261 6e73 6665 722d  00.OK..Transfer-
	0x0050:  456e 636f 6469 6e67 3a20 6368 756e 6b65  Encoding:.chunke
	0x0060:  640d 0a43 6f6e 7465 6e74 2d54 7970 653a  d..Content-Type:
	0x0070:  2074 6578 742f 6874 6d6c 3b20 6368 6172  .text/html;.char
	0x0080:  7365 743d 6973 6f2d 3838 3539 2d31 0d0a  set=iso-8859-1..
	0x0090:  0d0a                                     ..
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 55)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0xfe8e (correct), seq 93:108, ack 19, win 8192, length 15
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0037 0003 0000 4006 5cab 0a09 0807 0a01  .7....@.\.......
	0x0020:  0203 0050 bdec 0000 005e 0000 0014 5010  ...P.....^....P.
	0x0030:  2000 fe8e 0000 410d 0a41 4243 4445 4647  ......A..ABCDEFG
	0x0040:  4849 4a0d 0a                             HIJ..
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 49)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0xdc53 (correct), seq 108:117, ack 19, win 8192, length 9
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0031 0004 0000 4006 5cb0 0a09 0807 0a01  .1....@.\.......
	0x0020:  0203 0050 bdec 0000 006d 0000 0014 5010  ...P.....m....P.
	0x0030:  2000 dc53 0000 340d 0a45 4647 480d 0a    ...S..4..EFGH..
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xb2fa (correct), ack 117, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 5cb9 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0014 0000 0076 5010  .....P.......vP.
	0x0030:  2000 b2fa 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 49)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0xd442 (correct), seq 117:126, ack 19, win 8192, length 9
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0031 0005 0000 4006 5caf 0a09 0807 0a01  .1....@.\.......
	0x0020:  0203 0050 bdec 0000 0076 0000 0014 5010  ...P.....v....P.
	0x0030:  2000 d442 0000 340d 0a49 4a4b 4c0d 0a    ...B..4..IJKL..
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 51)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0xc222 (correct), seq 126:137, ack 19, win 8192, length 11
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0033 0006 0000 4006 5cac 0a09 0807 0a01  .3....@.\.......
	0x0020:  0203 0050 bdec 0000 007f 0000 0014 5010  ...P..........P.
	0x0030:  2000 c222 0000 340d 0a4d 4e4f 500d 0a0d  ..."..4..MNOP...
	0x0040:  0a                                       .
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xb2e6 (correct), ack 137, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 5cb8 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0014 0000 008a 5010  .....P........P.
	0x0030:  2000 b2e6 0000                           ......
IP (tos 0x0, ttl 64, id 7, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [F.], cksum 0xb2e5 (correct), seq 137, ack 19, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0007 0000 4006 5cb6 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0050 bdec 0000 008a 0000 0014 5011  ...P..........P.
	0x0030:  2000 b2e5 0000                           ......
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [F.], cksum 0xb2e4 (correct), seq 19, ack 138, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0006 0000 4006 5cb7 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0014 0000 008b 5011  .....P........P.
	0x0030:  2000 b2e4 0000                           ......
IP (tos 0x0, ttl 64, id 8, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0xb2e4 (correct), ack 20, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0008 0000 4006 5cb5 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0050 bdec 0000 008b 0000 0015 5010  ...P..........P.
	0x0030:  2000 b2e4 0000                           ......
